src/figuresClasses.ts

```diff
@@ -1,11 +1,81 @@
-export interface Figure {}
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
-export class Triangle implements Figure {}
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  shape: Shape = 'triangle';
+
+  constructor(
+    public color: Color,
+    public sideA: number,
+    public sideB: number,
+    public sideC: number,
+  ) {
+    if (
+      sideA >= sideB + sideC ||
+      sideB >= sideA + sideC ||
+      sideC >= sideA + sideB
+    ) {
+      throw new Error(
+        'The sum of the lengths of any two sides is greater than a third.',
+      );
+    }
+  }
+
+  getArea(): number {
+    const p = (this.sideA + this.sideB + this.sideC) / 2;
+    const s = Math.sqrt(
+      p * (p - this.sideA) * (p - this.sideB) * (p - this.sideC),
+    );
+
+    return Math.floor(s * 100) / 100;
+  }
+}
+
+export class Circle implements Figure {
+  shape: Shape = 'circle';
 
-export class Circle implements Figure {}
+  constructor(
+    public color: Color,
+    public radius: number,
+  ) {
+    if (radius <= 0) {
+      throw new Error('Radius must be greater than 0.');
+    }
+  }
 
-export class Rectangle implements Figure {}
+  getArea(): number {
+    const s = Math.PI * this.radius * this.radius;
+
+    return Math.floor(s * 100) / 100;
+  }
+}
+
+export class Rectangle implements Figure {
+  shape: Shape = 'rectangle';
+
+  constructor(
```

Comment:
The constructor parameters are being manually assigned to class properties, which is unnecessary when using TypeScript's parameter properties with access modifiers. You can remove the explicit assignments and the class will still have the properties 'color', 'sideA', and 'sideB'.